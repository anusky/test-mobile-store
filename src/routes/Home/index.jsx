import { useState } from "react";
import ProductList from "src/components/ProductList";
import SearchBar from "src/components/SearchBar";
import { useProductContext } from "src/context/productContext";

const Home = () => {
  const { list } = useProductContext();
  const [currentList, setCurrentList] = useState(list);

  const handleOnUpdateProductList = (filteredProductList) => {
    setCurrentList(filteredProductList);
  };
  return (
    <div data-testid="home-page-component" className="container mx-auto px-4">
      <SearchBar
        initialProductList={list}
        handleOnUpdateProductList={handleOnUpdateProductList}
      />
      <ProductList list={currentList} />
    </div>
  );
};

export default Home;
