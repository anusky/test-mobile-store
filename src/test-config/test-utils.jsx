import "./__mocks__/intersectionObserverMock";

import { render } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import PropTypes from "prop-types";

import productInfo from "./__fixtures__/product.json";
import productList from "./__fixtures__/productList.json";
import cart from "./__fixtures__/cart.json";
import { ProductContextProvider } from "src/context/productContext";
import { PAGE_STRUCTURE } from "src/utils/constants";

const customRender = (ui, options = {}) => {
  const { routerInitialIndex = 0 } = options;

  const AllTheProviders = ({ children }) => {
    const routes = [
      {
        path: "/",
        element: children,
        id: PAGE_STRUCTURE.home.clientPageId,
        loader: () => ({
          productList,
          cart,
        }),
        handle: {
          crumb: () => ({ url: "/", content: "home route" }),
        },
        children: [
          {
            path: "/:productId",
            element: children,
            id: PAGE_STRUCTURE.product.clientPageId,
            loader: () => productInfo,
            handle: {
              crumb: () => ({ url: "/", content: "product route" }),
            },
          },
        ],
      },
    ];

    const router = createMemoryRouter(routes, {
      initialEntries: ["/", "/123"],
      initialIndex: routerInitialIndex,
    });
    return (
      <RouterProvider router={router}>
        <ProductContextProvider
          value={{
            list: productList,
            cart: cart,
          }}
        >
          {children}
        </ProductContextProvider>
      </RouterProvider>
    );
  };

  AllTheProviders.propTypes = {
    children: PropTypes.node.isRequired,
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
