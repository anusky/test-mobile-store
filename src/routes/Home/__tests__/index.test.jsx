import userEvent from "@testing-library/user-event";
import { act } from "react-dom/test-utils";
import { render, screen, waitFor } from "src/test-config/test-utils";
import Home from "..";

describe("Home coverage", () => {
  it("Home renders properly", async () => {
    render(<Home />);
    expect(
      await screen.findByTestId("home-page-component")
    ).toBeInTheDocument();
  });

  it("matches snapshot", async () => {
    let container;
    await act(() => {
      container = render(<Home />);
    });
    expect(container.asFragment()).toMatchSnapshot();
  });
});
