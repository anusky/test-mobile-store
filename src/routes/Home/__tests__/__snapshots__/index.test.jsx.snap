// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home coverage matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto px-4"
    data-testid="home-page-component"
  >
    <div
      class="grid md:inline-flex text-gray-600 gap-4 place-items-center"
    >
      <div
        class="relative mx-auto w-full md:w-auto"
      >
        <input
          aria-label="Search for an smartphone"
          class="border-2 border-gray-300 bg-white h-10 px-5 pr-10 rounded-lg text-sm focus:outline-none w-[inherit]"
          placeholder="Search by brand,model"
          type="search"
        />
        <div
          class="absolute right-3 top-0 bottom-0 grid place-items-center"
        >
          <svg
            class="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M9 3.5a5.5 5.5 0 100 11 5.5 5.5 0 000-11zM2 9a7 7 0 1112.452 4.391l3.328 3.329a.75.75 0 11-1.06 1.06l-3.329-3.328A7 7 0 012 9z"
              fill-rule="evenodd"
            />
          </svg>
        </div>
      </div>
      <span
        aria-label="Total search results"
      />
    </div>
    <div
      class="grid grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-y-8 justify-center"
      data-testid="productlist-component"
    >
      <div />
    </div>
  </div>
</DocumentFragment>
`;
